drivers/mipi_dsi/dsi_mcux_2l.c

```diff
@@ -0,0 +1,248 @@
+
+	if (config->dphy_ref_freq != 0) {
+		dphy_bit_clk_freq = DSI_InitDphy(config->base,
+					&dphy_config, config->dphy_ref_freq);
+	} else {
+		/* DPHY PLL is not present, ref clock is unused */
+		DSI_InitDphy(config->base, &dphy_config, 0);
+	}
+
+	/* Init DPI interface. */
+	DSI_SetDpiConfig(config->base, &config->dpi_config, mdev->data_lanes,
+					dsi_pixel_clk_freq, dphy_bit_clk_freq);
+
+	imxrt_post_init_display_interface();
+
+	return 0;
+}
+
+static ssize_t dsi_mcux_transfer(const struct device *dev, uint8_t channel,
+				 struct mipi_dsi_msg *msg)
+{
+	const struct display_mcux_mipi_dsi_config *config = dev->config;
+	dsi_transfer_t dsi_xfer = {0};
+	status_t status;
+
+	dsi_xfer.virtualChannel = channel;
+	dsi_xfer.txDataSize = msg->tx_len;
+	dsi_xfer.txData = msg->tx_buf;
+	dsi_xfer.rxDataSize = msg->rx_len;
+	dsi_xfer.rxData = msg->rx_buf;
+
+	switch (msg->type) {
+
+	case MIPI_DSI_DCS_READ:
+		LOG_ERR("DCS Read not yet implemented or used");
+		return -ENOTSUP;
+	case MIPI_DSI_DCS_SHORT_WRITE:
+		dsi_xfer.sendDscCmd = true;
+		dsi_xfer.dscCmd = msg->cmd;
+		dsi_xfer.txDataType = kDSI_TxDataDcsShortWrNoParam;
+		break;
+	case MIPI_DSI_DCS_SHORT_WRITE_PARAM:
+		__fallthrough;
+	case MIPI_DSI_DCS_LONG_WRITE:
+		dsi_xfer.sendDscCmd = true;
+		dsi_xfer.dscCmd = msg->cmd;
+		dsi_xfer.txDataType = kDSI_TxDataDcsShortWrOneParam;
+		break;
+	case MIPI_DSI_GENERIC_SHORT_WRITE_0_PARAM:
+		dsi_xfer.txDataType = kDSI_TxDataGenShortWrNoParam;
+		break;
+	case MIPI_DSI_GENERIC_SHORT_WRITE_1_PARAM:
+		dsi_xfer.txDataType = kDSI_TxDataGenShortWrOneParam;
+		break;
+	case MIPI_DSI_GENERIC_SHORT_WRITE_2_PARAM:
+		dsi_xfer.txDataType = kDSI_TxDataGenShortWrTwoParam;
+		break;
+	case MIPI_DSI_GENERIC_READ_REQUEST_0_PARAM:
+		__fallthrough;
+	case MIPI_DSI_GENERIC_READ_REQUEST_1_PARAM:
+		__fallthrough;
+	case MIPI_DSI_GENERIC_READ_REQUEST_2_PARAM:
+		LOG_ERR("Generic Read not yet implemented or used");
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Unsupported message type (%d)", msg->type);
+		return -ENOTSUP;
+	}
+
+	status = DSI_TransferBlocking(config->base, &dsi_xfer);
+
+	if (status != kStatus_Success) {
+		LOG_ERR("Transmission failed");
+		return -EIO;
+	}
+
+	if (msg->rx_len != 0) {
+		/* Return rx_len on a read */
+		return msg->rx_len;
+	}
+
+	/* Return tx_len on a write */
+	return msg->tx_len;
+
+}
+
+static struct mipi_dsi_driver_api dsi_mcux_api = {
+	.attach = dsi_mcux_attach,
+	.transfer = dsi_mcux_transfer,
+};
+
+static int display_mcux_mipi_dsi_init(const struct device *dev)
+{
+	const struct display_mcux_mipi_dsi_config *config = dev->config;
+
+	imxrt_pre_init_display_interface();
+
+	if (!device_is_ready(config->bit_clk_dev) ||
+			!device_is_ready(config->esc_clk_dev) ||
+			!device_is_ready(config->pixel_clk_dev)) {
+		return -ENODEV;
+	}
+	return 0;
+}
+
+#define MCUX_MIPI_DSI_DEVICE(id)								\
+	static const struct display_mcux_mipi_dsi_config display_mcux_mipi_dsi_config_##id = {	\
```

Comment:
Please avoid this very long name, specially for variables with very limited scope, just
`static const struct mcux_mipi_dsi_config mipi_dsi_config_##id`.

Reply:
Fixed in lasted push